.eslintrc.cjs

```diff
@@ -1,39 +1,6 @@
 module.exports = {
-  env: {
-    browser: true,
-    es2021: true,
-    node: true,
-  },
-  extends: ['eslint:recommended', 'plugin:import/recommended'],
-  overrides: [
-    {
-      files: ['**/*.test.js', 'test/**/*.js'],
-      env: {
-        // technically, we are using vitest, but that's pretty similar to jest
-        jest: true,
-      },
-      settings: {
-        'import/ignore': [
-          // for some reason, `import {it} from "vitest";` throws an error
-          /node_modules\/vitest\/dist\/index\.js$/.source,
-        ],
-      },
-    },
-  ],
-  parserOptions: {
-    sourceType: 'module',
-  },
-  rules: {
-    'import/no-cycle': 'error',
-    // make sure that all files have an extension (required by ESM)
-    'import/extensions': [
-      'error',
-      'always',
-      {
-        js: 'always',
-        jsx: 'never',
-        mjs: 'always',
-      },
-    ],
-  },
```

Comment:
Instead of changing the file completely, like what you've done here, I'd prefer just adding the `plugin:@typescript-eslint/recommended`. I've also disabled the `'import/no-unresolved'` rule since it doesn't seem to work, and TypeScript should check this for us anyway:

```typescript
module.exports = {
  env: {
    browser: true,
    es2021: true,
    node: true,
  },
  extends: ['eslint:recommended', 'plugin:import/recommended', 'plugin:@typescript-eslint/recommended'],
  parser: '@typescript-eslint/parser',
  overrides: [
    {
      files: ['**/*.test.js', 'test/**/*.js'],
      env: {
        // technically, we are using vitest, but that's pretty similar to jest
        jest: true,
      },
      settings: {
        'import/ignore': [
          // for some reason, `import {it} from "vitest";` throws an error
          /node_modules\/vitest\/dist\/index\.js$/.source,
        ],
      },
    },
  ],
  parserOptions: {
    sourceType: 'module',
  },
  rules: {
    'import/no-cycle': 'error',
    'import/no-unresolved': 'off', // TypeScript's Node16 rule will check this for us
    // make sure that all files have an extension (required by ESM)
    'import/extensions': [
      'error',
      'always',
      {
        js: 'always',
        jsx: 'never',
        mjs: 'always',
      },
    ],
  },
};
```